cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -133,7 +133,10 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
 
     if key_supports_outlines(course_key):
         # Push the course outline to learning_sequences asynchronously.
-        update_outline_from_modulestore_task.delay(course_key_str)
+        update_outline_from_modulestore_task.apply_async(
+            args=[course_key_str],
+            countdown=settings.BLOCK_STRUCTURES_SETTINGS['COURSE_PUBLISH_TASK_DELAY'],
+        )
```

Comment:
These changes solved our issue, which I explained in the cover letter. So, we were getting an out-of-date read from Mongo; therefore, after the 1st save, the course outline structure was created without the new changes. Now, the second save (a few seconds later) triggered a read with the latest changes, so the course outline structure was created with up-to-date data.

I'm not sure why the course outline is updated from the module store immediately after a course is published, causing out-of-date reads. I don't know either about the specific consequences of delaying the read a few seconds.

I'd prefer to extract these changes to another PR, since it's not directly related. I'll do that soon.

Reply:
This is the PR: https://github.com/openedx/edx-platform/pull/34020, I'll proceed to remove these changes.